app/src/main/java/tech/relaycorp/letro/awala/processor/AwalaMessageProcessor.kt

```diff
@@ -3,17 +3,37 @@ package tech.relaycorp.letro.awala.processor
 import tech.relaycorp.awaladroid.messaging.IncomingMessage
 import tech.relaycorp.letro.awala.AwalaManager
 import tech.relaycorp.letro.awala.message.MessageType
+import tech.relaycorp.letro.utils.Logger
 
 interface AwalaMessageProcessor {
+    suspend fun isFromExpectedSender(
+        message: IncomingMessage,
+        awalaManager: AwalaManager,
+    ): Boolean
     suspend fun process(message: IncomingMessage, awalaManager: AwalaManager)
 }
 
 class AwalaMessageProcessorImpl constructor(
     private val processors: Map<MessageType, AwalaMessageProcessor>,
+    private val logger: Logger,
 ) : AwalaMessageProcessor {
 
     override suspend fun process(message: IncomingMessage, awalaManager: AwalaManager) {
         val type = MessageType.from(message.type)
-        processors[type]!!.process(message, awalaManager)
+        val processor = processors[type]!!
+        if (processor.isFromExpectedSender(message, awalaManager)) {
+            processors[type]!!.process(message, awalaManager)
+        } else {
+            logger.w(TAG, IllegalStateException("There is a message processor to process the message $type, but it came from unexpected sender"))
```

Comment:
The `IllegalStateException` is very misleading -- if I see it in the logs, I'd think that was actually thrown. Why don't we just log the message?

I think the message could be a bit more succinct and specific:

```suggestion
            logger.w(TAG, "Refusing message $type because it came from unexpected sender (${message.sender.nodeId})")
```

Reply:
Good spot! I just added method with this signature in `Logger` class in a different branch, and forgot to fix it here.

Reply:
Done